spec.bs

```diff
@@ -1146,10 +1146,36 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
 1. If |seller| is failure, then return failure.
 1. Set |auctionConfig|'s [=auction config/seller=] to |seller|.
 1. If |config|["{{AuctionAdConfig/serverResponse}}"] [=map/exists=]:
-  1. If |config|["{{AuctionAdConfig/requestId}}"] does not [=map/exist=], then [=exception/throw=]
-    a {{TypeError}}.
-  1. Set |auctionConfig|'s [=auction config/server response=] to |config|["{{AuctionAdConfig/serverResponse}}"].
-  1. Set |auctionConfig|'s [=auction config/server response id=] to |config|["{{AuctionAdConfig/requestId}}"].
+  1. If |config|["{{AuctionAdConfig/requestId}}"] does not [=map/exist=], then [=exception/throw=] a
+    {{TypeError}}.
+  1. Let |resolvedAndTypeChecked| be the promise representing performing the following steps
+    [=upon fulfillment=] of |config|["{{AuctionAdConfig/serverResponse}}"] with |result|:
+    1. Set |auctionConfig|'s [=auction config/server response=] to the [=ArrayBuffer/transferred=]
+      |config|["{{AuctionAdConfig/serverResponse}}"].

```

Comment:
The RHS of the assignment should be |result|. 

...Also it's unclear what's doing the type conversion to make sure |result| is a proper type.... Though that's probably underspec'd in WebIDL itself --- I had some conversations with a colleague making Chrome's WebIDL generator pay attention to the T in Promise<T> and it was unclear as to where the conversion is supposed to happen in the WebIDL layer.

Reply:
I may be wrong on the last part:
https://webidl.spec.whatwg.org/#js-promise:~:text=To%20react%20to%20a%20Promise%3CT%3E%20promise%2C%20given%20one%20or%20two%20sets%20of%20steps%20to%20perform%2C%20covering%20when%20the%20promise%20is%20fulfilled%2C%20rejected%2C%20or%20both%2C%20perform%20the%20following%20steps%3A

Reply:
... So just using |result| should be fine, and it should be a Uint8Array.

Reply:
done.